src/model/validation.rs

```diff
@@ -0,0 +1,424 @@
+//! Logic for model ([`Manifest`]) validation
+//!
+
+use std::collections::HashMap;
+use std::path::Path;
+use std::sync::OnceLock;
+
+use anyhow::{Context as _, Result};
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+
+use crate::model::{LinkProperty, Manifest, TraitProperty, LATEST_VERSION};
+
+/// A namespace -> package -> interface lookup
+type KnownInterfaceLookup = HashMap<String, HashMap<String, HashMap<String, ()>>>;
+
+/// Hard-coded list of known namespaces/packages and the interfaces they contain.
+///
+/// Using an interface that is *not* on this list is not an error --
+/// custom interfaces are expected to not be on this list, but when using
+/// a known namespace and package, interfaces should generally be well known.
+static KNOWN_INTERFACE_LOOKUP: OnceLock<KnownInterfaceLookup> = OnceLock::new();
+
+/// Get the static list of known interfaces
+fn get_known_interface_lookup() -> &'static KnownInterfaceLookup {
+    KNOWN_INTERFACE_LOOKUP.get_or_init(|| {
+        HashMap::from([
+            (
+                "wrpc".into(),
+                HashMap::from([
+                    (
+                        "blobstore".into(),
+                        HashMap::from([("blobstore".into(), ())]),
+                    ),
+                    ("config".into(), HashMap::from([("runtime".into(), ())])),
+                    (
+                        "keyvalue".into(),
+                        HashMap::from([("atomics".into(), ()), ("store".into(), ())]),
+                    ),
+                    (
+                        "http".into(),
+                        HashMap::from([("incoming-handler".into(), ())]),
+                    ),
+                ]),
+            ),
+            (
+                "wasi".into(),
+                HashMap::from([
+                    (
+                        "blobstore".into(),
+                        HashMap::from([("blobstore".into(), ())]),
+                    ),
+                    ("config".into(), HashMap::from([("runtime".into(), ())])),
+                    (
+                        "keyvalue".into(),
+                        HashMap::from([("atomics".into(), ()), ("store".into(), ())]),
+                    ),
+                    (
+                        "http".into(),
+                        HashMap::from([("incoming-handler".into(), ())]),
+                    ),
+                    ("logging".into(), HashMap::from([("logging".into(), ())])),
+                ]),
+            ),
+            (
+                "wasmcloud".into(),
+                HashMap::from([
+                    (
+                        "bus".into(),
+                        HashMap::from([("lattice".into(), ()), ("host".into(), ())]),
+                    ),
```

Comment:
This one should just be a builtin, so we shouldn't specify a link in the manifest right?

Reply:
Ah true -- we haven't used this in a dynamic way at all so far, will remove